### PR TITLE
Add L0-based tools for GPU monitoring

### PR DESCRIPTION
**Pull request by rlnx**
Introduce L0-based tool to get information about GPU :fire:

**Build:**
```sh
bazel build //dev/l0_tools:zeinfo
```

**Usage:**
```sh
# Info about GPUs
./bazel-bin/dev/l0_tools/zeinfo
# or 
./bazel-bin/dev/l0_tools/zeinfo info

# Dynamically monitor frequency
./bazel-bin/dev/l0_tools/zeinfo freq
```

In general, you may find out LevelZero API usage examples useful in the future.

Example of tool output:
```sh
[GPU:0] XXXXXXXXXXXXXXXXXXX [xxxxxx] 
   Threads per EU: X
   EU SIMD width (32-bit lane): X
   EUs layout (EUs x sub-slices x slices): X x X x X = XX
   Peak f32_mad/clock: XXX
   Min frequency (core): XXXMHz
   Max frequency (core): XXXXMHz
```

### CODE REVIEW

File: dev/l0_tools/zexx.hpp [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        std::vector<frequency_domain> domains;
+        domains.reserve(domain_handles.size());
+        for (auto handle : domain_handles) {
+            domains.emplace_back(handle);
+        }
+        return domains;
+    }
+
+    void query_frequency() const {
+        // zesDeviceEnumFrequencyDomains()
```
Comment (vepifanov) [inline dev/l0_tools/zexx.hpp:282]: Is it left empty intentionally?

File: dev/l0_tools/utils.hpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+template <typename Container>
+class enumerate_iterator {
```
Comment (KulikovNikita) [inline dev/l0_tools/utils.hpp:20]: So why don't you inherit properties from std::iterator?

File: dev/l0_tools/zeinfo.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    return monitors;
+}
+
+void display_frequency_monitors() {
+    constexpr char move_cursor_up[] = "\033[A\033[K";
+
+    const auto monitors = create_frequency_monitors();
+    while (true) {
```
Comment (KulikovNikita) [inline dev/l0_tools/zeinfo.cpp:132]: What about checking if user pressed any key?

**Review by KulikovNikita — APPROVED** [review]

File: dev/l0_tools/zeinfo.cpp (hunk 3 above)
Comment (rlnx) [reply dev/l0_tools/zeinfo.cpp:132]: You always have `Ctrl` + `C`

File: dev/l0_tools/utils.hpp (hunk 2 above)
Comment (rlnx) [reply dev/l0_tools/utils.hpp:20]: `std::iterator` is deprecated in C++17

**Pull request merged**